### DISCUSSION TITLE
Update Firmware

### DISCUSSION THREAD

**Discussion by tzureig** [mriscoc/Ender3V2S1#1173] [General]
I would like to update the firmware from one Professional firmware to another, e.g. from version 20230904 to version 20231202. How can I retain my data? I have already tried a few things, but nothing has worked. Thank you very much for your help.

**Comment by TheWebMachine** [comment]
You'll do this via config file: you'll write up this file, place on your SD or save to Octoprint and run it every time you load new FW to reload your settings. It's super convenient, as most of the commands are Marlin native and will work on most other Marlin firmwares out there.

https://github.com/mriscoc/Ender3V2S1/wiki/Configuration-files

**Reply by tzureig** [reply]
Thanks for the help
